hudi-common/src/main/java/org/apache/hudi/common/table/timeline/TimelineUtils.java

```diff
@@ -210,11 +210,30 @@ public static HoodieDefaultTimeline getTimeline(HoodieTableMetaClient metaClient
     return activeTimeline;
   }
 
+  /**
+   * Returns a Hudi timeline with commits after the given instant time (exclusive).
+   *
+   * @param metaClient                {@link HoodieTableMetaClient} instance.
+   * @param exclusiveStartInstantTime Start instant time (exclusive).
+   * @return Hudi timeline.
+   */
+  public static HoodieTimeline getCommitsTimelineAfter(
+      HoodieTableMetaClient metaClient, String exclusiveStartInstantTime) {
+    HoodieActiveTimeline activeTimeline = metaClient.getActiveTimeline();
+    HoodieDefaultTimeline timeline =
+        activeTimeline.isBeforeTimelineStarts(exclusiveStartInstantTime)
+            ? metaClient.getArchivedTimeline(exclusiveStartInstantTime)
+            .mergeTimeline(activeTimeline)
+            : activeTimeline;
+    return timeline.getCommitsTimeline()
+        .findInstantsAfter(exclusiveStartInstantTime, Integer.MAX_VALUE);
+  }
```

Comment:
Only the active timeline needs to be filtered by the start instant time, i.e. the invocation `#findInstantsAfter`

Reply:
We need to scan all the instants since `exclusiveStartInstantTime` to figure out the touched partitions and it is possible that `exclusiveStartInstantTime` is before the start of the archived timeline, in which case we need to still scan the archived timeline (see #6662 for details).  In most of the cases, `exclusiveStartInstantTime` should be after the start of the active timeline, so the archived timeline is not loaded.

Reply:
I mean if `activeTimeline.isBeforeTimelineStarts(exclusiveStartInstantTime)` is true, the whole merged timeline should be scanned, there is no need to calling `#findInstantsAfter`.

Reply:
Is this resolved ?

Reply:
I think I misunderstood your comment before.  When `activeTimeline.isBeforeTimelineStarts(exclusiveStartInstantTime)` is true, `metaClient.getArchivedTimeline(exclusiveStartInstantTime)` is called to return the archived timeline, which still contains the instant exclusiveStartInstantTime (inclusive).  So we still need `#findInstantsAfter` as the last step to exclude that instant.

Reply:
@yihua I have a doubt, since rollback and commit are archived separately, is it possible that there is a very early rollback instant, causing `activeTimeline.isBeforeTimelineStarts(exclusiveStartInstantTime)` to return false?